### PR TITLE
Cross-Compiler Documentation Update + Minor CodeQL Alert Resolutions

### PR DESCRIPTION
**Pull request by Boehm-Michael**
| | |
|:---|:---|
|**_Originating Project/Creator_**| boehm-Michael |
|**_Affected Component_**| N/A |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**| CodeQL |
|**_Has Unit Tests (y/n)_**| N/A |
|**_Builds Without Errors (y/n)_**| Y |
|**_Unit Tests Pass (y/n)_**| N/A |
|**_Documentation Included (y/n)_**| Y |

---
## Change Description

Updated cross-compiler documentation and fixed code readability as noted by CodeQL

## Rationale

Resolve CodeQL alerts, fix broken cross-compiler documentation

## Testing/Review Recommendations

N/A

## Future Work

Continue to update for CodeQL alerts.

### CODE REVIEW

**Comment by Boehm-Michael** [comment]
Spell Check is hanging on PowerShell code block. Alerts -DisplayName is not a word because it interprets it as isplayName, removing the '-D'. I'm assuming it thinks it's a flag. There are a few of these being alerted on.

**Comment by thomas-bc** [comment]
@jsoref tagging you in case you can help, no pressure at all.

We are having spell-check report Powershell flags like `-DisplayName` and `-Direction`, interpreting them as `isplayName` and `irection` (see [here](https://github.com/nasa/fprime/pull/2096/files#diff-bfc0ac53d029a7868d2a6c6230f9eb8eab1eaa0d4aa226b70f2b923a0eaca5b7R35)). Is there a way to prevent/resolve that without adding these incorrect spellings to expect.txt?
Thanks!

**Comment by jsoref** [comment]
@thomas-bc https://github.com/nasa/fprime/blob/4f65e1ef3fa3aa44e132e6d9e7b885e840162630/.github/actions/spelling/patterns.txt#L5 is what's causing it.

I think you could probably change it to:
```
(?:^|[\t ,"'`=(])-(?:[DP](?=[A-Z]{2,})|[WXYLlf](?=[A-Z]{2,}|[A-Z][a-z]|[a-z]{2,}))
```

**Comment by jsoref** [comment]
You'll want to comment (`#`) out https://github.com/nasa/fprime/blob/4f65e1ef3fa3aa44e132e6d9e7b885e840162630/.github/actions/spelling/candidate.patterns#L515 so that check-spelling doesn't suggest restoring the pattern.

**Comment by thomas-bc** [comment]
@jsoref Awesome, thank you very much! 🎉

File: .github/actions/spelling/patterns.txt [hunk 1]
```diff
@@ -91,6 +91,7 @@ value="(?:[0-9a-f]{1,2} )*"
 
 # Ignore any text between inline back-ticks
 `(.*?)`
+```(.*?)```
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt]: This won't do what you're expecting... check-spelling doesn't currently support multi-line concepts, and this pattern would be covered by the inline back-ticks pattern.

I'm still thinking about the steps necessary to handle multiline patterns (the current file really won't support it, so I basically need to imagine a new file and a new syntax for the file, or something) -- https://github.com/check-spelling/check-spelling/wiki/Feature%3A-Block-Ignore

Comment (thomas-bc) [reply .github/actions/spelling/patterns.txt]: Understood, thank you for the explanation. I have indeed removed it from the PR I opened cause I noticed it didn't work.

File: .github/actions/spelling/patterns.txt [hunk 2]
```diff
@@ -128,4 +128,4 @@ GithubProjectProperty
 \b([A-Za-z])\g{-1}{3,}\b
 
 # ignore docker platform paths
-\--platform=\(linux\|darwin\)/\(amd64\|arm\|arm32v5\|arm32v6\|arm32v7\|arm64v8\|i386\|ppc64le\|s390x\|x86_64\)
\ No newline at end of file
+--platform=(linux|darwin)/(amd64|arm|arm32v5|arm32v6|arm32v7|arm64v8|i386|ppc64le|s390x|x86_64)
```
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:131]: if you don't really care, this is probably better:
```suggestion
--platform=\w+/\w+
```

**Pull request closed without merging**